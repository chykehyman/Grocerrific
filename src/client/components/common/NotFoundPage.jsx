import React from 'react';
import PropTypes from 'prop-types';


const propTypes = {
  history: PropTypes.shape({
    location: PropTypes.shape({
      pathname: PropTypes.string.isRequired
    }).isRequired
  }).isRequired
};

const NotFoundPage = (props) => {
  const style = { fontWeight: 'bold' };
  const { history: { location } } = props;
  const message = ` ${location.pathname} `;

  return (
    <div className="no-items">
      <p className="lead mb-4 mt-4 display-4 text-secondary">404</p>
      <i className="fa fa-exclamation-triangle fa-3x pb-3 d-block text-secondary" />
      <p className="lead text-secondary">
        Ooops!!!.. Page with url
        <span style={style}>{message}</span>
        was not found
      </p>
    </div>
  );
};

NotFoundPage.propTypes = propTypes;

export default NotFoundPage;
