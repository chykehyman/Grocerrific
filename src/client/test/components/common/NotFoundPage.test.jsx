import NotFoundPage from '../../../components/common/NotFoundPage';


const props = {
  history: {
    location: {
      pathname: ''
    }
  }
};

describe('<NotFoundPage /> component test', () => {
  it('renders 404(not found page) component without crashing', () => {
    const shallowWrapper = shallow(<NotFoundPage {...props} />);
    expect(toJson(shallowWrapper)).toMatchSnapshot();
  });
});
