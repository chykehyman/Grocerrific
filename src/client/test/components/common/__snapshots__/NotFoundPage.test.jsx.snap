// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NotFoundPage /> component test renders 404(not found page) component without crashing 1`] = `
<div
  className="no-items"
>
  <p
    className="lead mb-4 mt-4 display-4 text-secondary"
  >
    404
  </p>
  <i
    className="fa fa-exclamation-triangle fa-3x pb-3 d-block text-secondary"
  />
  <p
    className="lead text-secondary"
  >
    Ooops!!!.. Page with url
    <span
      style={
        Object {
          "fontWeight": "bold",
        }
      }
    >
        
    </span>
    was not found
  </p>
</div>
`;
